homeassistant/components/dlink/config_flow.py

```diff
@@ -19,6 +20,25 @@
 class DLinkFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for D-Link Power Plug."""
 
+    def __init__(self) -> None:
+        """Initialize a D-Link Power Plug flow."""
+        self.ip_address: str | None = None
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle dhcp discovery."""
+        await self.async_set_unique_id(discovery_info.macaddress)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: discovery_info.ip})
+        self.ip_address = discovery_info.ip
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.data[CONF_HOST] == discovery_info.ip and not entry.unique_id:
```

Comment:
```suggestion
            if not entry.unique_id and entry.data[CONF_HOST] == discovery_info.ip: 
```

MINOR: I'd reverse this conditional since a `not` is usually cheaper than a string comparison.